bundles/io/org.eclipse.smarthome.io.security/src/main/java/org/eclipse/smarthome/io/security/api/SecretsVaultService.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.security.api;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Offers to store secrets and retrieve secrets again. Those secrets can
+ * only be retrieved by the bundle that stored them and will be encrypted
+ * and saved in the user directory of the framework.
+ *
+ * The encryption algorithm and storage format depends on the actual implementation.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public interface SecretsVaultService {
+    /**
+     * Retrieve a secret, identified by the given key.
+     *
+     * @param key An identifier
+     * @return Either null if no secret could be found under the given key, or the secret.
+     * @throws IOException If the underlying file can't be opened or read.
+     * @throws GeneralSecurityException If the decryption algorithm fails, this exception is thrown
+     */
+    @Nullable
+    ByteBuffer getSecret(@NonNull String key) throws IOException, GeneralSecurityException;
+
+    /**
+     * Set or delete a secret, identified by the given key.
+     *
+     * @param key An identifier
+     * @param secret A secret as a ByteBuffer. Can be null to delete the secret.
```

Comment:
Addition: If the vault previously contained a value for the key, the old value is replaced.